js/index.ts

```diff
@@ -0,0 +1,241 @@
+/**
```

Comment:
In a pedantic way, now that you've moved to typescript (nice!), the root should be 'src' or something vs 'js'. But that might all pain no gain.

Reply:
Should be easy enough and it's always annoyed me that it wasn't `src`. Thanks for the excuse to change it!